### PR TITLE
Fixed verbose in QC, integrated coveralls

### LINKED ISSUE #1772: "open file" does not work on OSX
### Description

At the end of a processing, the following command display is issued for users to open a browser and glance at QC results:
~~~
open file "/Users/julien/qc_batch_processing/index.html"
~~~

However, this command fails on OSX and needs to be replaced with:
~~~
open "/Users/julien/qc_batch_processing/index.html"
~~~

### State of spinalcordtoolbox

Spinal Cord Toolbox (jca_batchProc/dea98c961553315202c90086fddedf923e99a768)

### LINKED ISSUE #1785: ssh without display: add test to make sure SCT does not crash
To avoid #1784

### PR DESCRIPTION
**Pull request by abtahizadeh**
### Requirements

* [x] Is there a issue open describing the motivation for this pull request ?
#1772 
* [x] Does the pull request need an update of the documentation ?
Yes, coveralls.io integrated (icon appears in MarkDown
* [ ] Does the pull request need additional test(s) ?
* [ ] Clean up the warnings generated by the [style guide](https://www.jetbrains.com/help/pycharm/2017.1/code-inspection.html)

### Description of the Change
- Fixing the issue by detecting the OS,
- Initial coverage integration
- Minor pep8 format changes

### Steps and Constraints

[The steps of reproducing the pull request]

### Applicable Issues

Fixes #1772, Fixes #1785

### CODE REVIEW

File: spinalcordtoolbox/reports/qc.py [hunk 1]
```diff
@@ -413,4 +419,13 @@ def normalized(img):
 
     sct.printv('Sucessfully generated the QC results in %s' % qc_param.qc_results)
     sct.printv('Use the following command to see the results in a browser:')
-    sct.printv('open file "{}/index.html"'.format(path_qc), type='info')
+    try:
```
Comment (zougloub) [inline spinalcordtoolbox/reports/qc.py]: The try/except is overkill.

File: spinalcordtoolbox/reports/qc.py [hunk 2]
```diff
@@ -413,4 +419,13 @@ def normalized(img):
...
+        from sys import platform as _platform
```
Comment (zougloub) [inline spinalcordtoolbox/reports/qc.py]: sys is already imported earlier.

File: spinalcordtoolbox/reports/qc.py [hunk 3]
```diff
@@ -413,4 +419,13 @@ def normalized(img):
...
+        if _platform == "linux" or _platform == "linux2":
+            sct.printv('xdg-open "{}/index.html"'.format(path_qc), type='info')
+        elif _platform == "darwin":
+            sct.printv('open "{}/index.html"'.format(path_qc), type='info')
+        else:
```
Comment (zougloub) [inline spinalcordtoolbox/reports/qc.py]: We could bail out here (`RuntimeError`). Then we will remember to fix that when we  want to support another platform.

File: spinalcordtoolbox/reports/qc.py (hunk 1 above)
Comment (abtahizadeh) [reply spinalcordtoolbox/reports/qc.py]: ImportError raises if the OS is not detectable

File: spinalcordtoolbox/reports/qc.py (hunk 3 above)
Comment (abtahizadeh) [reply spinalcordtoolbox/reports/qc.py]: (ImportError, RuntimeError) would do the job?

File: spinalcordtoolbox/reports/qc.py (hunk 2 above)
Comment (abtahizadeh) [reply spinalcordtoolbox/reports/qc.py]: catching the exception in case of a non-detectable OS;

**Comment by abtahizadeh** [comment]
Fix for issue #1785 added

> ssh without display: add test to make sure SCT does not crash

Retrying with setting env variable, printing relevant message if failed.

**Pull request merged**